themes/default/content/blog/converting-full-terraform-programs-to-pulumi/index.md

```diff
@@ -0,0 +1,184 @@
+---
+title: "Converting Full Terraform Programs to Pulumi"
+date: 2023-06-12
+meta_desc: Learn how to convert whole Terraform programs to Pulumi using the new Terraform converter
+
+
+meta_image: meta.png
+
+authors:
+    - justin-vanpatten
+
+tags:
+    - migration
+---
+
+Today, we're excited to announce support for converting whole Terraform projects to Pulumi via the `pulumi convert` command in the Pulumi CLI. The new Terraform converter includes support for Terraform modules, all core features of Terraform 1.4, and the majority of Terraform built-in functions. Let's dig in to learn more about the new converter and how to use it.
+
+<!--more-->
+
+Historically, we have offered a separate [tf2pulumi](https://github.com/pulumi/tf2pulumi) tool to convert snippets of Terraform to Pulumi. Today we're announcing a new converter that supports converting the vast majority of complete Terraform projects, including modules. There's no longer a separate tool to download -- you can run the new converter directly from the Pulumi CLI with the `pulumi convert --from terraform` command to convert Terraform to Pulumi TypeScript, Python, Go, or C#.
+
+## Supported Terraform Features
+
+The following major features are supported:
+
+* Variables, providers, outputs, resources, and data sources
+* Terraform Modules are converted to Pulumi components
+* Almost all HCL2 expression syntax
+* Almost all Terraform meta-arguments are supported
```

Comment:
Ehh this feels dishonest, we're missing most of them currently. I think it's only count and range that we currently handle. Easy to add to the converter, harder to add to PCL and we're missing engine support for triggers.

Reply:
Removed this bullet